### PR TITLE
Workspaced

### PR DESCRIPTION
**Pull request by Murasame233**
make project into workspace.

- Proxmox-encode: for encoding data to Proxmox format without breaking json format.
- tests: for testing

### CODE REVIEW

File: tests/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[package]
```
Comment (datdenkikniet) [inline tests/Cargo.toml]: I would prefer if these were tests under `proxmox-api/tests` in the way Cargo [recommends running integration tests](https://doc.rust-lang.org/rust-by-example/testing/integration_testing.html)

Comment (Murasame233) [reply tests/Cargo.toml]: all right, I ll delete the tests crate

**Comment by Murasame233** [comment]
Alright, I think every thing's ok now.

**Comment by datdenkikniet** [comment]
I would merge, but it seems that there are merge conflicts. Could you solve them, please?

**Comment by Murasame233** [comment]
> I would merge, but it seems that there are merge conflicts. Could you solve them, please?

I will working on it on Monday, I am not at home

**Comment by Murasame233** [comment]
resolvd

**Comment by datdenkikniet** [comment]
CI is failing: seems like a merge gone bad. Could you please `git merge` or `git rebase` so we get proper history, and a passing CI run

Edit: but also, it seems like there may have been some changes to proxmox itself. Gonna fix that now!

**Comment by datdenkikniet** [comment]
I've resolved the problem with the JSON schema being a new version (and logged #17 because apparently the version of `proxmox-ve` is not specific enough), but it seems that this PR is missing some `Cargo.lock` updates :)

**Comment by Murasame233** [comment]
Give me some time, let me see what's going on.

**Comment by datdenkikniet** [comment]
Just running `cargo check` on all the projects should update the respective `Cargo.lock` files.

**Comment by Murasame233** [comment]
I think if you check the cargo workspace documentation, you run `Cargo check` on root directory, It will generate a `Cargo.lock` in root directory. And we don't need `Cargo.lock` on sub directory.

If you want to run a sub crate of the workspaces
```
cargo run -p <package name>
```


ref: https://doc.rust-lang.org/book/ch14-03-cargo-workspaces.html

**Comment by datdenkikniet** [comment]
Would you like to fix it or should I do it? I cannot push commits to your branch, so I'd have to do it in a separate PR

**Comment by Murasame233** [comment]
> Would you like to fix it or should I do it? I cannot push commits to your branch, so I'd have to do it in a separate PR

Sorry for the delay, I have a lot issue in my reality life. Now all resolved.
Now I will working for the pr I want to.
You can do it in separate PR, and I will continue the encoding pr.

**Comment by Murasame233** [comment]
I force pushed this pr, Maybe there's no conflict now.

File: Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+[workspace]
+members = ["generator", "proxmox-api"]
```
Comment (datdenkikniet) [inline Cargo.toml:2]:
```suggestion
members = ["generator", "proxmox-api"]
resolver = "2"
```

There's some warning about which resolver version to use, please add this

**Review by datdenkikniet — CHANGES_REQUESTED** [review]

**Comment by datdenkikniet** [comment]
> Sorry for the delay, I have a lot issue in my reality life. Now all resolved. Now I will working for the pr I want to. You can do it in separate PR, and I will continue the encoding pr.

No worries, good to hear that you resolved it :) Please address my last comment, and we should (finally :P) be good to go!

**Review by datdenkikniet — APPROVED** [review]

**Comment by datdenkikniet** [comment]
Merged! Thank you for your patience :)

**Pull request merged**